src/Markdig/Extensions/AutoLinks/AutoLinkParser.cs

```diff
@@ -182,21 +185,23 @@ public override bool Match(InlineProcessor processor, ref StringSlice slice)
             if (Options.OpenInNewWindow)
             {
                 inline.GetAttributes().AddPropertyIfNotExist("target", "blank");
-            }
-
-            return true;
+            }
+
+            _listOfCharCache.Release(pendingEmphasis);

```

Comment:
Wondering if a plain try/finally wouldn't be easier in that case and avoid the goto?

Reply:
Haven't thought of that one. I suppose it would be cleaner since there are many exit points. I'll make the change.